code/game/objects/effects/overlays.dm

```diff
@@ -25,18 +25,25 @@
 	mouse_opacity = 0
 	var/tmp/atom/BeamSource
 
-/obj/effect/overlay/beam/New(turf/loc, var/lifetime = 10)
+/obj/effect/overlay/beam/New(turf/loc, var/lifetime = 10, var/fade = 0)
 	..()
 	spawn()
 		sleep(lifetime/4)
-		alpha = 192
+		if(fade)
+			alpha = 192
 		sleep(lifetime/4)
-		alpha = 128
+		if(fade)
+			alpha = 128
 		sleep(lifetime/4)
-		alpha = 64
+		if(fade)
+			alpha = 64

```

Comment:
I've never used animate before, but now that I check the documentation that looks like it would work, and even look better.